DOCSP-34486: v6.3 release (#828)
mongodb/docs-node: source/whats-new.txt

* :ref:`Version 3.7 <version-3.7>`
* :ref:`Version 3.6 <version-3.6>`

.. _version-6.3:

What's New in 6.3
-----------------

The {+driver-short+} v6.3 release includes the following features:

- Adds the ``serverMonitoringMode`` client option to control the

>   behavior of the monitoring connection between each node in a topology.

Reviewer: I: [between/among](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#:~:text=between-,When%20referring%20to,-three%20or%20more)

```suggestion
  behavior of the monitoring connection among the nodes [or "between each pair of nodes"; i'm not sure which is correct] in a topology.
```

  This option takes a value of ``auto`` (default), ``poll``, or

>   ``stream``. When the setting is ``auto``, the environment the driver
>   runs in determines the monitoring mode. Function-as-a-service (FaaS)

Reviewer: S: I would push for passive voice here for clarity

```suggestion
  ``stream``. When the setting is ``auto``, the monitoring mode is
   determined by the environment where the driver is running. Function-as-a-service (FaaS)
```

>   runs in determines the monitoring mode. Function-as-a-service (FaaS)
>   environments prefer the polling mode and all others prefer the
>   streaming mode.

Reviewer: S: "prefer" doesn't seem quite right and is pretty [anthropomorphic](https://www.mongodb.com/docs/meta/style-guide/writing/use-effective-verbs/#don-t-humanize-inanimate-objects)

```suggestion
  runs in determines the monitoring mode. The driver uses polling
  mode in function-as-a-service (FaaS)
  environments and streaming mode everywhere else.
```

  You can set the ``serverMonitoringMode`` option in a
  ``MongoClientOptions`` instance or as a connection string option. The
  following example shows how to create a client with the option set to
  ``stream``:
  
  .. code-block:: js
     

>      new MongoClient('<connection string>', { serverMonitoringMode: 'stream' });

Reviewer: S: I would probably leave most or all of this info to be added in the docs ticket for this feature, but it doesn't hurt to leave it in.

Author: I can take this out of the whats new and add it to the connection options entry (in this PR as well) instead.

- Fixes a connection leak when the ``serverApi`` client option is set.
  In previous versions, the driver sent a legacy command to the server
  that returned an error, and the error-handling logic didn't correctly
  remove the socket causing the connection leak.

> - Deprecates the ``contentType`` and ``aliases`` GridFS options. To
>   store this information, add corresponding fields to the ``metadata``
>   document.

Reviewer: S: clarity

```suggestion
- Deprecates the ``contentType`` and ``aliases`` GridFS options. To
  store the content type and aliases, add ``contentType`` and ``aliases``
  fields to the ``metadata`` document.
```
